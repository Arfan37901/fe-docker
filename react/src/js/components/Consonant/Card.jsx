import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Tooltip from './Tooltip';

const CARD_STYLE = {
    WIDE: '3:2',
    SQUARE: '1:1',
    FULL: 'full-card',
};

const Card = (props) => {
    const {
        id,
        title,
        label,
        description,
        ctaLink,
        ctaLabel,
        image,
        bannerDescription,
        bannerFontColor,
        bannerBackgroundColor,
        bannerIcon,
        secondaryLabelText,
        allowBookmarking,
        isBookmarked,
        cardSavedIco,
        cardUnsavedIco,
        onClick,
        saveBookmarkText,
        unsaveBookmarkText,
        cardStyle,
        videoURL,
    } = props;

    const handleClick = (clickEvt) => {
        clickEvt.stopPropagation();
        onClick(id);
    };

    const setClassName = () => {
        const res = ['consonant-card'];
        const style = cardStyle.toLowerCase().trim();

        if (style === CARD_STYLE.FULL) res.push('consonant-card_full');
        if (style === CARD_STYLE.SQUARE) res.push('consonant-card_square');

        return res.join(' ');
    };

    return (
        <div className={setClassName()} id={id}>
            <div
                className="consonant-card--img"
                style={{ backgroundImage: `url("${image}")` }}>
                {
                    bannerDescription &&
                    bannerFontColor &&
                    bannerBackgroundColor &&
                    <span
                        className="consonant-card--banner"
                        style={({
                            backgroundColor: bannerBackgroundColor,
                            color: bannerFontColor,
                        })}>
                        {
                            bannerIcon &&
                            <img
                                src={bannerIcon}
                                alt=""
                                loading="lazy"
                                width="8"
                                className="consonant-card--banner-icon" />
                        }
                        {bannerDescription}
                    </span>
                }
                { videoURL && <span className="consonant-card--video-ico" />}
            </div>
            <div className="consonant-card--inner">
                {
                    label &&
                    <span className="consonant-card--label">{label}</span>
                }
                <h2
                    className="consonant-card--title"
                    dangerouslySetInnerHTML={{ __html: title }} />
                {
                    ![CARD_STYLE.SQUARE, CARD_STYLE.FULL]
                        .some(el => el === cardStyle.toLowerCase().trim()) &&
                        <Fragment>
                            <p
                                className="consonant-card--text"
                                dangerouslySetInnerHTML={{ __html: description }} />
                            <div className="consonant-card--footer-wrapper">
                                <div className="consonant-card--footer-info">
                                    {
                                        allowBookmarking &&
                                        <button
                                            data-tooltip-wrapper
                                            type="button"
                                            className={
                                                isBookmarked ?
                                                    'consonant-card--footer-btn consonant-card--footer-btn_active' :
                                                    'consonant-card--footer-btn'
                                            }
                                            onClick={handleClick}>
                                            {(cardSavedIco && cardUnsavedIco) ?
                                                <img
                                                    src={
                                                        isBookmarked ?
                                                            cardSavedIco :
                                                            cardUnsavedIco
                                                    }
                                                    width="16"
                                                    alt=""
                                                    loading="lazy" /> :
                                                <span className="consonant-card--bookmark-ico" />
                                            }
                                            <Tooltip text={
                                                isBookmarked ? unsaveBookmarkText : saveBookmarkText
                                            } />
                                        </button>

                                    }
                                    {
                                        secondaryLabelText &&
                                        <span className="consonant-card--secondary-text">{secondaryLabelText}</span>
                                    }
                                </div>
                                <a
                                    href={ctaLink}
                                    target="_blank"
                                    className="consonant-card--btn">{ctaLabel}
                                </a>
                            </div>
                        </Fragment>
                }
            </div>
        </div>
    );
};

export default Card;

Card.propTypes = {
    id: PropTypes.string.isRequired,
    ctaLink: PropTypes.string.isRequired,
    ctaLabel: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    label: PropTypes.string,
    description: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    bannerDescription: PropTypes.string,
    bannerFontColor: PropTypes.string,
    bannerBackgroundColor: PropTypes.string,
    bannerIcon: PropTypes.string,
    secondaryLabelText: PropTypes.string,
    allowBookmarking: PropTypes.bool.isRequired,
    isBookmarked: PropTypes.bool.isRequired,
    cardSavedIco: PropTypes.string.isRequired,
    cardUnsavedIco: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    saveBookmarkText: PropTypes.string,
    unsaveBookmarkText: PropTypes.string,
    cardStyle: PropTypes.string,
    videoURL: PropTypes.string,
};

Card.defaultProps = {
    bannerIcon: '',
    bannerDescription: '',
    bannerFontColor: '',
    bannerBackgroundColor: '',
    secondaryLabelText: '',
    label: '',
    saveBookmarkText: 'Save card',
    unsaveBookmarkText: 'Unsave card',
    cardStyle: '',
    videoURL: '',
};
