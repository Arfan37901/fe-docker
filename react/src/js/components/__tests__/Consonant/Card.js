import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Card from '../../Consonant/Card';

import {
    CARD_STYLE,
    DEFAULT_PROPS,
    PROPS_WITHOUT_BANNER,
} from '../constants/card';

import makeSetup from '../utils/settings';

Enzyme.configure({ adapter: new Adapter() });

const getSetup = makeSetup(Card, DEFAULT_PROPS);

describe('Consonant/Card', () => {
    test('should renders banner correctly', () => {
        const {
            props,
            wrapper,
        } = getSetup();

        const bannerElement = wrapper.find('.consonant-card--banner');
        const bannerIconElement = wrapper.find('.consonant-card--banner-icon');

        expect(bannerElement).toHaveLength(1);
        expect(bannerElement.get(0).props.style).toHaveProperty('color', props.bannerFontColor);
        expect(bannerElement.get(0).props.style).toHaveProperty('backgroundColor', props.bannerBackgroundColor);

        expect(bannerElement.text()).toEqual(props.bannerDescription);

        expect(bannerIconElement.get(0).props.src).toEqual(props.bannerIcon);
    });
    test('shouldn`t render banner if some of banner prop dont exists', () => {
        PROPS_WITHOUT_BANNER.forEach((propsWithoutBanner) => {
            const { wrapper } = getSetup(propsWithoutBanner);

            const bannerElement = wrapper.find('.consonant-card--banner');

            expect(bannerElement).toHaveLength(0);
        });
    });
    test('should renders without banner icon', () => {
        const { wrapper } = getSetup({ bannerIcon: '' });

        const bannerIconElement = wrapper.find('.consonant-card--banner-icon');

        expect(bannerIconElement).toHaveLength(0);
    });
    test('should renders without video icon', () => {
        const { wrapper } = getSetup({ videoURL: '' });

        const videoIconElement = wrapper.find('.consonant-card--video-ico');

        expect(videoIconElement).toHaveLength(0);
    });
    test('should renders without label', () => {
        const { wrapper } = getSetup({ label: '' });

        const cardLabelElement = wrapper.find('.consonant-card--label');

        expect(cardLabelElement).toHaveLength(0);
    });

    describe('Check snapshots', () => {
        test('should renders with bookmarking', () => {
            const { tree } = getSetup({
                isBookmarked: true,
                allowBookmarking: true,
                disableBookmarkIco: true,
            });

            expect(tree).toMatchSnapshot();
        });
        test('should renders correctly with different card style', () => {
            CARD_STYLE.forEach((style) => {
                const { tree } = getSetup({ cardStyle: style });

                expect(tree).toMatchSnapshot();
            });
        });
    });

    describe('Interaction with UI', () => {
        test('should call onClick', () => {
            const { props: { onClick }, wrapper } = getSetup({ allowBookmarking: true, cardStyle: '3:2' });

            wrapper.find('button').simulate('click', { stopPropagation: () => {} });

            expect(onClick).toBeCalled();
        });
    });
});
