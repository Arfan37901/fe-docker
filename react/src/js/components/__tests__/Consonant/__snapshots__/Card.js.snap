// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Card Check snapshots should renders correctly with different card style 1`] = `
<div
  className="consonant-card"
  id="6d6438c8-1"
>
  <div
    className="consonant-card--img"
    style={
      Object {
        "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
      }
    }
  >
    <span
      className="consonant-card--banner"
      style={
        Object {
          "backgroundColor": "#CB6F10",
          "color": "#FFFFFF",
        }
      }
    >
      <img
        alt=""
        className="consonant-card--banner-icon"
        loading="lazy"
        src="some-banner-icon"
        width="8"
      />
      Some Banner Description
    </span>
    <span
      className="consonant-card--video-ico"
    />
  </div>
  <div
    className="consonant-card--inner"
  >
    <span
      className="consonant-card--label"
    >
      Some label
    </span>
    <h2
      className="consonant-card--title"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Some Title 1 ",
        }
      }
    />
    <p
      className="consonant-card--text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Some Description",
        }
      }
    />
    <div
      className="consonant-card--footer-wrapper"
    >
      <div
        className="consonant-card--footer-info"
      >
        <button
          className="consonant-card--footer-btn"
          data-tooltip-wrapper={true}
          onClick={[Function]}
          type="button"
        >
          <span
            className="consonant-card--bookmark-ico"
          />
          <span
            className="consonant-tooltip"
          >
            Save breakout session
          </span>
        </button>
        <span
          className="consonant-card--secondary-text"
        >
          Some secondary text
        </span>
      </div>
      <a
        className="consonant-card--btn"
        href="https://google.com"
        rel="noreferrer"
        target="_blank"
      >
        Some CTA
      </a>
    </div>
  </div>
</div>
`;

exports[`Consonant/Card Check snapshots should renders correctly with different card style 2`] = `
<div
  className="consonant-card consonant-card_square"
  id="6d6438c8-1"
>
  <div
    className="consonant-card--img"
    style={
      Object {
        "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
      }
    }
  >
    <span
      className="consonant-card--banner"
      style={
        Object {
          "backgroundColor": "#CB6F10",
          "color": "#FFFFFF",
        }
      }
    >
      <img
        alt=""
        className="consonant-card--banner-icon"
        loading="lazy"
        src="some-banner-icon"
        width="8"
      />
      Some Banner Description
    </span>
    <span
      className="consonant-card--video-ico"
    />
  </div>
  <div
    className="consonant-card--inner"
  >
    <span
      className="consonant-card--label"
    >
      Some label
    </span>
    <h2
      className="consonant-card--title"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Some Title 1 ",
        }
      }
    />
  </div>
</div>
`;

exports[`Consonant/Card Check snapshots should renders correctly with different card style 3`] = `
<div
  className="consonant-card"
  id="6d6438c8-1"
>
  <div
    className="consonant-card--img"
    style={
      Object {
        "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
      }
    }
  >
    <span
      className="consonant-card--banner"
      style={
        Object {
          "backgroundColor": "#CB6F10",
          "color": "#FFFFFF",
        }
      }
    >
      <img
        alt=""
        className="consonant-card--banner-icon"
        loading="lazy"
        src="some-banner-icon"
        width="8"
      />
      Some Banner Description
    </span>
    <span
      className="consonant-card--video-ico"
    />
  </div>
  <div
    className="consonant-card--inner"
  >
    <span
      className="consonant-card--label"
    >
      Some label
    </span>
    <h2
      className="consonant-card--title"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Some Title 1 ",
        }
      }
    />
    <p
      className="consonant-card--text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Some Description",
        }
      }
    />
    <div
      className="consonant-card--footer-wrapper"
    >
      <div
        className="consonant-card--footer-info"
      >
        <button
          className="consonant-card--footer-btn"
          data-tooltip-wrapper={true}
          onClick={[Function]}
          type="button"
        >
          <span
            className="consonant-card--bookmark-ico"
          />
          <span
            className="consonant-tooltip"
          >
            Save breakout session
          </span>
        </button>
        <span
          className="consonant-card--secondary-text"
        >
          Some secondary text
        </span>
      </div>
      <a
        className="consonant-card--btn"
        href="https://google.com"
        rel="noreferrer"
        target="_blank"
      >
        Some CTA
      </a>
    </div>
  </div>
</div>
`;

exports[`Consonant/Card Check snapshots should renders correctly with different card style 4`] = `
<div
  className="consonant-card consonant-card_full"
  id="6d6438c8-1"
>
  <div
    className="consonant-card--img"
    style={
      Object {
        "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
      }
    }
  >
    <span
      className="consonant-card--banner"
      style={
        Object {
          "backgroundColor": "#CB6F10",
          "color": "#FFFFFF",
        }
      }
    >
      <img
        alt=""
        className="consonant-card--banner-icon"
        loading="lazy"
        src="some-banner-icon"
        width="8"
      />
      Some Banner Description
    </span>
    <span
      className="consonant-card--video-ico"
    />
  </div>
  <div
    className="consonant-card--inner"
  >
    <span
      className="consonant-card--label"
    >
      Some label
    </span>
    <h2
      className="consonant-card--title"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Some Title 1 ",
        }
      }
    />
  </div>
</div>
`;

exports[`Consonant/Card Check snapshots should renders with bookmarking 1`] = `
<div
  className="consonant-card"
  id="6d6438c8-1"
>
  <div
    className="consonant-card--img"
    style={
      Object {
        "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
      }
    }
  >
    <span
      className="consonant-card--banner"
      style={
        Object {
          "backgroundColor": "#CB6F10",
          "color": "#FFFFFF",
        }
      }
    >
      <img
        alt=""
        className="consonant-card--banner-icon"
        loading="lazy"
        src="some-banner-icon"
        width="8"
      />
      Some Banner Description
    </span>
    <span
      className="consonant-card--video-ico"
    />
  </div>
  <div
    className="consonant-card--inner"
  >
    <span
      className="consonant-card--label"
    >
      Some label
    </span>
    <h2
      className="consonant-card--title"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Some Title 1 ",
        }
      }
    />
    <p
      className="consonant-card--text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Some Description",
        }
      }
    />
    <div
      className="consonant-card--footer-wrapper"
    >
      <div
        className="consonant-card--footer-info"
      >
        <button
          className="consonant-card--footer-btn consonant-card--footer-btn_active consonant-card--footer-btn_disabled"
          data-tooltip-wrapper={true}
          onClick={[Function]}
          type="button"
        >
          <span
            className="consonant-card--bookmark-ico"
            style={
              Object {
                "backgroundImage": "url(https://upload.wikimedia.org/wikipedia/commons/a/a0/Ukrposhta-ua-icon.svg)",
              }
            }
          />
          <span
            className="consonant-tooltip"
          >
            Unsave breakout session
          </span>
        </button>
        <span
          className="consonant-card--secondary-text"
        >
          Some secondary text
        </span>
      </div>
      <a
        className="consonant-card--btn"
        href="https://google.com"
        rel="noreferrer"
        target="_blank"
      >
        Some CTA
      </a>
    </div>
  </div>
</div>
`;
